pkg/cloudprovider/aws.go

```diff
@@ -348,3 +345,26 @@ func (a *AWS) getInstance(node *corev1.Node) (*ec2.Instance, error) {
 	}
 	return instances[0], nil
 }
+
+// getInstanceIdFromProviderId extracts the instance id from a given provider id.
+// The provider ID should normally be passed in with aws:///<zone>/<instanceID>
+// But it might also be passed in as aws://<zone>/<instanceID> and the zone might
+// be omitted.
+// This is what the node's providerID can look like on AWS (examples):
+//    spec:
+//      providerID: aws:///us-west-2a/i-008447f243eead273
+//      providerID: aws://us-west-2a/i-008447f243eead273
+//      providerID: aws:///i-008447f243eead273
+// see https://github.com/kubernetes/cloud-provider-aws/blob/5f394ba297bf280ceb3edfc38922630b4bd83f46/pkg/providers/v2/instances.go#L254
+func getInstanceIdFromProviderId(providerId string) (string, error) {
+	// after trimming 'aws://', the remainder will be in one of the following formats:
+	// * <availability-zone>/<instance-id>
+	// * /<availability-zone>/<instance-id>
+	// * <instance-id>
+	splitProviderId := strings.Split(strings.TrimPrefix(providerId, "aws://"), "/")
+	if len(splitProviderId) < 1 || len(splitProviderId) > 3 {
```

Comment:
`splitProviderId` will always have at least one element, https://pkg.go.dev/strings#Split:
`If s does not contain sep and sep is not empty, Split returns a slice of length 1 whose only element is s. `

Reply:
Addressed, thanks

Reply:
and no, it was just a mistake. Should I check for an empty providerID, though?

Reply:
ok I added a check for that as well, PTAL :-)